docs/contributing.rst

```diff
@@ -1 +1,190 @@
+
+   $ make coverage
+
+#. Push your changes to GitHub and submit a Pull Request. The base branch should be a new feature branch that we create to receive the changes (contact us to create the branch). This allows us to test the changes using our build system before merging to main.
+
+.. note:: 
+
+    We enable typeguard with pytest so if you add type declarations to your code, those declarations will be treated as runtime assertions in your Python tests.
+
+Pull Request Guidelines
+-----------------------
+
+Before you submit a pull request, check that it meets these guidelines:
+
+* Documentation is updated to reflect all changes.
+* New classes, functions, etc have docstrings.
+* New code has comments.
+* Code style and file structure is similar to the rest of the project.
+* You have run the ``black`` code formatter.
+* If you are a new contributor, don't forget to add yourself to the ``AUTHORS.rst`` file in your pull request (either GitHub username, or first/last name).
+
+Testing CumulusCI
+-----------------
+
+Org-reliant Automated Tests
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Some tests are marked ``@pytest.mark.vcr()`` which means that they can either
+call into a real (configured) Salesforce org or use a cached YAML file of the request/response.
+
+By default using pytest will use the cached YAML. If you want to work against a
+real scratch org, you do so like this::
+
+    $ pytest --org qa <other arguments and options, such as filename or -k testname>
+
+Where "orgname" is a configured org name like "qa", "dev", etc.
+
+To regenerate the VCR file, you can run this command::
+
+    $ pytest --replace-vcrs --org qa
+
+This will configure an org named "qa" and regenerate them.
+
+That will run all VCR-backed tests against the org, including all of the slow
+integration tests.
+
+Running Integration Tests
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Some tests generate so much data that we do not want to store the VCR cassettes
+in our repo. You can mark tests like that with ``@pytest.mark.large_vcr()``. When
+they are executed, their cassettes will go in a .gitignore'd folder called
+``large_cassettes``.
+
+Do not commit the files (``large_cassettes/\*.yml``) to the repository.
+
+Some tests generate even more network traffic data and it isn't practical 
+to use VCR at all. Still, we'd like to run them when we run all of the
+other org-reliant tests with --org. Mark them with ``@pytest.mark.needs_org()``
+and they will run with the VCR tests.
+
+Some tests are so slow that you only want to run them on an opt-in basis.
+Mark these tests with ``@pytest.mark.slow()`` and run them with
+``pytest --run-slow-tests`` or ``pytest --run-slow-tests --orgname <orgname>``.
+
+Writing Integration Tests
+~~~~~~~~~~~~~~~~~~~~~~~~~
+All features should have integration tests which work against
+real orgs or APIs.
+
+You will need to use some the following fixtures in your tests. Search
+the repo to see examples where they are used in context, or to see
```

Comment:
Some of these fixtures, like `project_config` have multiple definitions. May be worth mentioning.